nat-lab/tests/utils/connection_tracker.py

```diff
@@ -128,21 +175,67 @@ async def execute(self) -> None:
 
         await self._process.execute(stdout_callback=self.on_stdout)
 
+    async def get_all_events(self) -> List[ConntrackerEvent]:
+        """Returns a list of the reported events that matched the ConnectionTrackerConfig"""
+        if platform.system() == "Darwin":
+            return []
+        if not self._config:
+            return []
+
+        await self._synchronize()
+
+        async with self._lock:
+            return self._events
```

Comment:
You are returning a reference to a mutable type which you protect with a lock. Later in count_events_with_tcp_state you access it without locking.